zio-http/shared/src/main/scala/zio/http/internal/FormState.scala

```diff
@@ -72,7 +86,12 @@ private[http] object FormState {
           case ClosingBoundary(_)       => BoundaryClosed(tree)
         }
 
-      } else if (crlf && (phase eq Phase.Part2)) {
+      } else if ((crlf || boundaryDetected) && (phase eq Phase.Part2)) {
+        if (boundaryDetected) {
+          val hyphen = "-".getBytes().head
+          buffer += hyphen
+          buffer += hyphen
```

Comment:
Much more performant this way:

```suggestion
          buffer += '-'
          buffer += '-'
```

Reply:
Thanks, I would like to keep `+=` because in all other places in this file it is used to append to a `ChunkBuilder`.

Reply:
Also, I think boxing is not a big concern here as this is only done once per request.